subprojects/diagnostics/src/main/groovy/org/gradle/api/tasks/diagnostics/BuildscriptDependencyReportTask.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.tasks.diagnostics;
+
+import org.gradle.api.artifacts.ConfigurationContainer;
+
+/**
+ * Displays the buildscript dependency tree for a project. An instance of this type is used when you
+ * execute the {@code buildscriptDependencies} task from the command-line.
+ */
+public class BuildscriptDependencyReportTask extends AbstractDependencyReportTask {
```

Comment:
This guy shouldn't extend AbstractDependencyReportTask. It would be better to extend DefaultTask and expose no configuration options. This might mean that some logic from ADRT needs to be factored out to be decoupled from Task so that it can be reused.